### PR TITLE
Add output validations

### PR DESCRIPTION
**Pull request by Yuangwang**
Add validations check that the output bundle generated is as expected

### CODE REVIEW

File: packages/@apphosting/adapter-angular/src/utils.ts [hunk 1]
```diff
@@ -161,39 +166,52 @@ function extractManifestOutput(output: string): string {
...
+}
+
+// Validate output directory includes all necessary parts
+export async function validateOutputDirectory(
+  outputBundleOptions: OutputBundleOptions,
+): Promise<void> {
+  if (
+    !(await fsExtra.exists(outputBundleOptions.outputDirectory)) ||
+    !(await fsExtra.exists(outputBundleOptions.browserDirectory)) ||
+    !(await fsExtra.exists(outputBundleOptions.serverFilePath))
```
Comment (sjjj986) [inline packages/@apphosting/adapter-angular/src/utils.ts:211]: Should we also verify the existence of bundleYamlPath

File: packages/@apphosting/adapter-nextjs/src/utils.ts [hunk 2]
```diff
@@ -109,3 +109,15 @@ async function generateBundleYaml(
...
   return;
 }
+
+// Validate output directory includes all necessary parts
+export async function validateOutputDirectory(
+  outputBundleOptions: OutputBundleOptions,
+): Promise<void> {
+  if (
+    !(await fsExtra.exists(outputBundleOptions.outputDirectory)) ||
+    !(await fsExtra.exists(outputBundleOptions.serverFilePath))
```
Comment (sjjj986) [inline packages/@apphosting/adapter-nextjs/src/utils.ts:119]: Same here should we check bundle.yaml exists?

File: packages/@apphosting/adapter-angular/src/utils.ts (hunk 1 above)
Comment (Yuangwang) [reply packages/@apphosting/adapter-angular/src/utils.ts]: Yes good call I added these checks

File: packages/@apphosting/adapter-nextjs/src/utils.ts (hunk 2 above)
Comment (Yuangwang) [reply packages/@apphosting/adapter-nextjs/src/utils.ts]: Added

**Review by sjjj986 — APPROVED** [review]

**Pull request merged**